### PR TITLE
cheapest museum item to donate

### PR DESCRIPTION
**Pull request by Lulonaut**
depends on #492

does not catch all armor variations but I am not willing to work around Hypixels stupid naming inconsistencies

### CODE REVIEW

**Comment by efefury** [comment]
so true

**Comment by jani270** [comment]
working version btw

**Comment by hannibal002** [comment]
Though efe is working on this? Is this a Collab?

**Comment by NopoTheGamer** [comment]
more than 1 person working on a pr :scream:

**Comment by efefury** [comment]
multiple people working on a PR :scream_cat:

**Comment by Lulonaut** [comment]
_should_ be ready now

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/inventory/MuseumCheapestItemOverlay.kt [hunk 1]
```diff
@@ -231,6 +231,23 @@ object MuseumCheapestItemOverlay {
                 if (itemsToDonate.none { it.internalNames == internalNames }) {
                     itemsToDonate.add(MuseumItem(displayName, internalNames, value, time))
                 }
+            } else if (stack.item is ItemDye && stack.itemDamage == 10) { //also check donated items
+                val name = stack.displayName.stripControlCodes()
+                println(name)
```
Comment (NopoTheGamer) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/inventory/MuseumCheapestItemOverlay.kt]: println(lunar)

Comment (NopoTheGamer) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/inventory/MuseumCheapestItemOverlay.kt]: nvm i shouldnt probably look at old commits only

**Review by jani270 — DISMISSED** [review]
Bugs: 
- After opening the museum for the first time the museum overlay is blank, this is fixed after opening the 3rd page
- The text "Visit all pages..." only appears on the 4th page 
^ Both of those bugs were tested in the Weapons menu
- Some Items show Unknown because they dont have a lowest bin. You might want to change that to cant find... 

Suggestion:
- Upon clicking a item we could show the recipe
- Use the texture from the minion helper (One of the buttons could be using raw craft cost instead of lb)
  Note: We could change the texture to not having the buttons on the texture directly, this requires a change in the minion overlay tho.
- Add a * to items that got a higher tier (like Tita drill x555 would show a * because there is an x655)
- Show the Overlay on the main page "Your Museum"

**Review by jani270 — COMMENTED** [review]
The tooltip acts pretty weird it dosent show on the ride side of your mouse
Minion Helper: https://i.imgur.com/4vnicZo.png
Museum: https://i.imgur.com/WYEKvrf.png

**Comment by jani270** [comment]
Crashed while hovering over an item with a unknown bin:
```
java.lang.ArrayIndexOutOfBoundsException: 5
	at io.github.moulberry.notenoughupdates.util.Utils.shortNumberFormat(Utils.java:348)
	at io.github.moulberry.notenoughupdates.util.Utils.shortNumberFormat(Utils.java:350)
	at io.github.moulberry.notenoughupdates.util.Utils.shortNumberFormat(Utils.java:350)
	at io.github.moulberry.notenoughupdates.util.Utils.shortNumberFormat(Utils.java:350)
	at io.github.moulberry.notenoughupdates.util.Utils.shortNumberFormat(Utils.java:350)
	at io.github.moulberry.notenoughupdates.util.Utils.shortNumberFormat(Utils.java:350)
	at io.github.moulberry.notenoughupdates.miscfeatures.inventory.MuseumCheapestItemOverlay.drawLines(MuseumCheapestItemOverlay.kt:212)
	at io.github.moulberry.notenoughupdates.miscfeatures.inventory.MuseumCheapestItemOverlay.onDrawBackground(MuseumCheapestItemOverlay.kt:101)

**Comment by Lulonaut** [comment]
should be ready now :clueless:

**Review by NopoTheGamer — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/MiscOverlays.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import io.github.moulberry.notenoughupdates.core.config.annotations.ConfigAccordionId;
 import io.github.moulberry.notenoughupdates.core.config.annotations.ConfigEditorAccordion;
 import io.github.moulberry.notenoughupdates.core.config.annotations.ConfigEditorBoolean;
+import io.github.moulberry.notenoughupdates.core.config.annotations.ConfigEditorColour;
```
Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/MiscOverlays.java:27]: why

**Review by hannibal002 — CHANGES_REQUESTED** [review]

**Comment by hannibal002** [comment]
image

This is really hard to read. Maybe render with different shader options?

**Comment by hannibal002** [comment]
Why does this option in the config exist

image

When there is already this toggle button in the gui?

image

**Comment by hannibal002** [comment]
Can we show this
image

like in the hypixel ah inventory?
image

**Comment by hannibal002** [comment]
Minion helper is using the yellow non-bold formatting for the last line.
image

Same as Hypixel itself:

image

Why are we inventing a new design there?

image

File: src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/MiscOverlays.java (hunk 2 above)
Comment (hannibal002) [reply src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/MiscOverlays.java]: I have fixed it already.

**Review by romangraef — APPROVED** [review]

**Review by romangraef — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]
:(

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**